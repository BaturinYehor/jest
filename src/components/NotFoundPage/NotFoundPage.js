import React from "react";

class NotFoundPage extends React.Component {

    render() {
        return (
            <div className='wrapper'>
                <div className="card-form">
                    <div className='text'>
                        <div>404 page not found</div>
                        <a href='/greeting'>To greeting form</a>
                        <div className="img"/>
                    </div>
                </div>
            </div>
        )
    }
}

export default NotFoundPage