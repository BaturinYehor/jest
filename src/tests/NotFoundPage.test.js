import renderer from "react-test-renderer";
import React from "react";
import NotFoundPage from "../components/NotFoundPage/NotFoundPage";
import {mount} from "enzyme/build";

let wrapper;

describe('NotFoundPage', () => {

    beforeEach(() => {
        wrapper = mount(<NotFoundPage/>);
    });

    it('Should match snapshot', () => {
        const component = renderer.create(<NotFoundPage/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should have correct warning text', () => {
        const expectedTest = "404 page not found";
        expect(wrapper.find('.text').get(0).props.children[0].props.children).toEqual(expectedTest);
    });

    it('Should have correct ref text', () => {
        const expectedTest = "To greeting form";
        expect(wrapper.find('.text').get(0).props.children[1].props.children).toEqual(expectedTest);
    });

    it('Ref should have correct url', () => {
        const url = "/greeting";
        expect(wrapper.find('.text').get(0).props.children[1].props.href).toEqual(url);
    });
});