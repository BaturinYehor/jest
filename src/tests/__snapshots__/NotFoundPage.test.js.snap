// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage Should match snapshot 1`] = `
<div
  className="wrapper"
>
  <div
    className="card-form"
  >
    <div
      className="text"
    >
      <div>
        404 page not found
      </div>
      <a
        href="/greeting"
      >
        To greeting form
      </a>
      <div
        className="img"
      />
    </div>
  </div>
</div>
`;
